contributors/devel/sig-node/separate-mount-namespace.md

```diff
@@ -0,0 +1,204 @@
+
+After much investigation and conversation on both the sig-node and sig-storage
+mailing lists, as well as within Red Hat's OpenShift communities, we have yet
+to turn up a use case that relies on this container-to-host propagation.
+
+### Testing implications
+
+With separate mount namespaces, it would be possible to run multiple instances
+of Kubernetes on a single host without the various mountpoints interfering with
+one another. There may be other inter-instance difficulties besides the mount
+namespace, but this does move in that direction without resorting to more
+complicated solutions like trying to run Kubernetes inside of containers.
+
+## Costs and Implications
+
+If a Kubernetes deployment intends to implement this mount-hiding mode of
+operation, there are a number of costs and implications that should be
+documented for system administrators.
+
+### Privileged pods
+
+Privileged pods that previously expected to have direct access to the host's
+root mount namespace (such as the one spawned by `oc debug`), will now be
+rooted in the new container-runtime mount namespace.  They will still have
+visibility into the host OS's mount points as granted by the normal volumeMount
+permissions, but any mounts made in these privileged pods will not by default
+propagate up to be visible by the Host OS. In order for shells or utilities in
+privileged pods like this to propagate mounts up to the host OS, they can enter
+the host OS namespace by running `nsenter -t 1 -m` (assuming they have the host
+OS's /proc filesystem mounted so they can reach the mount namespace of the real
+init process in /proc/1/ns/mnt)
+
+### Legacy Tools
+
+Any monitoring tool that relies on the ability to run in the host OS and have
+visibility of mountpoints created by Kubelet, the Container Runtime, or
+containers themselves, will need to enter the container mount namespace in
+order to see these mountpoints.  This could be done by:
+
+- Converting the legacy tool to be containerized
+- Altering the scripts that launch the legacy tool to run it inside the
+  container-mount namespace
+- Altering the legacy tool to be aware of the container-mount namespace and
+  enter it as needed to probe or interact with the container-specific
+  mountpoints
+- Providing a system-wide mechanism to revert the Kubernetes installation back
+  to the original "all mounts in the Host OS" mode of operation.
+
+### Administrative Access
+
+An administrator logged in to the host OS wishing to obtain information about
+the Kubernetes system for debugging or auditing purposes will need to be aware
+of the new mount namespace, and will need to run `nsenter` to view or interact
+with these mountpoints.
+
+## Implementation Suggestions
+
+This does not require a change to Kubelet, the Container Runtime, or any other
+existing component, but can be accomplished entirely through changes to the
+environment and init scripts which launch Kublete and the Container Runtime.
```

Comment:
would it be desirable to have end-to-end test configured this way so we know that this suggestion actually works and no new changes will break it? How hard would it be?

Reply:
Yes, that's a good idea.  I'd need to learn more about what existing environments there are that run the e2e tests before I could tell you how hard it would be; Do you have any pointers to where these environments are defined or set up?

Reply:
@ike-ma spent some time recently thinking how to enable swap for tests. Ike, do you have quick pointers on what's the bet way to customize e2e tests for the scenario in this PR?